src/main/kotlin/net/maizegenetics/phgv2/api/HaplotypeGraph.kt

```diff
@@ -77,25 +72,32 @@ class HaplotypeGraph(hvcfFiles: List<String>) {
      * Returns a hapId -> sample list map for the given ReferenceRange.
      * Returned Map<hapId, List<sampleName>>
      */
-    fun hapIdToSamples(range: ReferenceRange): Map<String, List<String>> {
+    fun hapIdToSamples(range: ReferenceRange, gameteId: Int = 0): Map<String, List<String>> {
```

Comment:
I do not think this will work very well. Since the user has to know how many gametes per sample and it is possible to construct a graph with some haploid and some diploid samples. If this is called with gamete = 1 and some hapIdList in line 80 only have a single item, then I think that generates an error. I think what is needed is to have a method that takes ReferenceRange as the only argument and returns a map of hapid -> List<Pair<sampleName, gameteId>>. There may be a better return data structure but that is how I see it.

Reply:
I agree with Peter.  We need to couple the sampleName and gameteIds and propagate them through like they would be the 'Taxon' in phgv1.  Alternatively to Peters suggestion we could make a simple data class which would hold the sampleName and the gameteId together.  A pair works just as well though, but a class would be more descriptive when using it.

Reply:
Sounds like a data class is needed. Should I do that now? Or are we moving forward with what I have?